sqlglot/transforms.py

```diff
@@ -229,12 +229,26 @@ def remove_target_from_merge(expression: exp.Expression) -> exp.Expression:
     return expression
 
 
+def remove_within_group_for_percentiles(expression: exp.Expression) -> exp.Expression:
+    if (
+        isinstance(expression, exp.WithinGroup)
+        and isinstance(expression.this, (exp.PercentileCont, exp.PercentileDisc))
+        and isinstance(expression.expression, exp.Order)
+    ):
+        quantile = expression.this.this
+        input_value = t.cast(exp.Ordered, expression.find(exp.Ordered)).this
```

Comment:
The cast was added to suppress a MyPy error. Since `expression.expression` is an `exp.Order`, it'll always contain an `exp.Ordered` expression for its `expressions` arg, so this should be safe to use.